### PR TITLE
Remove the session module, and add micronaut-session

### LINKED ISSUE #8152: Extract session to its own repository
### Issue description

see: https://github.com/micronaut-projects/micronaut-core/tree/4.0.x/session

### PR DESCRIPTION
**Pull request by timyates**

### CODE REVIEW

**Comment by timyates** [comment]
Can you point me to the Upgrade wiki/doc?  As this will mean anything that is using a session will need the dependency changing

**Comment by timyates** [comment]
@graemerocher Do we want to keep `session` here with an `api` dependency on `micronaut-session`?  That's what was done with `runtime`...however the session module is outside of micronaut-core so I'm not sure if that's something we want to do?

File: src/main/docs/guide/httpServer/sessions.adoc [hunk 1]
```diff
@@ -99,33 +99,33 @@ snippet::io.micronaut.docs.session.ShoppingControllerSpec[tags="add", indent=0]
 
 == Using @SessionValue
 
-Rather than explicitly injecting the api:session.Session[] into a controller method, you can instead use ann:session.annotation.SessionValue[]. For example:
+Rather than explicitly injecting the https://micronaut-projects.github.io/micronaut-session/snapshot/api/io/micronaut/session/Session.html[Session] into a controller method, you can instead use https://micronaut-projects.github.io/micronaut-session/snapshot/api/io/micronaut/session/annotation/SessionValue.html[@SessionValue]. For example:
```
Comment (sdelamo) [inline src/main/docs/guide/httpServer/sessions.adoc]: define `https://micronaut-projects.github.io/micronaut-session/snapshot/api` in https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/gradle.properties and uses it in the asciidoc.

**Review by sdelamo — CHANGES_REQUESTED** [review]
You have to document this breaking change in https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/src/main/docs/guide/appendix/breaks.adoc

**Comment by sdelamo** [comment]
> Can you point me to the Upgrade wiki/doc? As this will mean anything that is using a session will need the dependency changing

breaking changes are documented always in `breaks.adoc`

**Comment by timyates** [comment]
Done and added to `breaks.adoc`

Fixed a couple of incorrect dependency macro usages there as well

**Review by sdelamo — APPROVED** [review]

**Review by graemerocher — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**